chain/src/consume_unverified.rs

```diff
@@ -1,414 +1,261 @@
-            blk.header().number()
-        })
-    }
-
-    pub fn during_hardfork(&self, hardfork_switch: &HardForks) -> bool {
-        let hardfork_during_detach =
-            self.check_if_hardfork_during_blocks(hardfork_switch, &self.detached_blocks);
-        let hardfork_during_attach =
-            self.check_if_hardfork_during_blocks(hardfork_switch, &self.attached_blocks);
-
-        hardfork_during_detach || hardfork_during_attach
-    }
-
-    fn check_if_hardfork_during_blocks(
-        &self,
-        hardfork: &HardForks,
-        blocks: &VecDeque<BlockView>,
-    ) -> bool {
-        if blocks.is_empty() {
-            false
-        } else {
-            // This method assumes that the input blocks are sorted and unique.
-            let rfc_0049 = hardfork.ckb2023.rfc_0049();
-            let epoch_first = blocks.front().unwrap().epoch().number();
-            let epoch_next = blocks
-                .back()
-                .unwrap()
-                .epoch()
-                .minimum_epoch_number_after_n_blocks(1);
-            epoch_first < rfc_0049 && rfc_0049 <= epoch_next
-        }
-    }
-}
-
-pub(crate) struct GlobalIndex {
-    pub(crate) number: BlockNumber,
-    pub(crate) hash: Byte32,
-    pub(crate) unseen: bool,
+    stop_rx: Receiver<()>,
+    processor: ConsumeUnverifiedBlockProcessor,
 }
 
-impl GlobalIndex {
-    pub(crate) fn new(number: BlockNumber, hash: Byte32, unseen: bool) -> GlobalIndex {
-        GlobalIndex {
-            number,
-            hash,
-            unseen,
+impl ConsumeUnverifiedBlocks {
+    pub(crate) fn new(
+        shared: Shared,
+        unverified_blocks_rx: Receiver<UnverifiedBlock>,
+        truncate_block_rx: Receiver<TruncateRequest>,
+        proposal_table: ProposalTable,
+        is_pending_verify: Arc<DashSet<Byte32>>,
+        stop_rx: Receiver<()>,
+    ) -> Self {
+        ConsumeUnverifiedBlocks {
+            tx_pool_controller: shared.tx_pool_controller().to_owned(),
+            unverified_block_rx: unverified_blocks_rx,
+            truncate_block_rx,
+            stop_rx,
+            processor: ConsumeUnverifiedBlockProcessor {
+                shared,
+                is_pending_verify,
+                proposal_table,
+            },
         }
     }
 
-    pub(crate) fn forward(&mut self, hash: Byte32) {
-        self.number -= 1;
-        self.hash = hash;
-    }
-}
-
-/// Chain background service
-///
-/// The ChainService provides a single-threaded background executor.
-pub struct ChainService {
-    shared: Shared,
-    proposal_table: ProposalTable,
-}
-
-impl ChainService {
-    /// Create a new ChainService instance with shared and initial proposal_table.
-    pub fn new(shared: Shared, proposal_table: ProposalTable) -> ChainService {
-        ChainService {
-            shared,
-            proposal_table,
-        }
-    }
+    pub(crate) fn start(mut self) {
+        loop {
+            let _trace_begin_loop = minstant::Instant::now();
```

Comment:
```suggestion
            let trace_begin_loop = minstant::Instant::now();
```

since it's used by later code.